ts/private/ts_project_worker.js

```diff
@@ -492,85 +597,47 @@ function createProgram(args, initialInputs) {
         }
     }
 
-    function debuglog(message) {
-        // TODO: https://github.com/aspect-build/rules_ts/issues/189
-        compilerOptions.extendedDiagnostics && host.trace?.(message)
-    }
-
     function readFile(filePath, encoding) {
-        const relative = path.relative(execRoot, filePath);
+        filePath = path.resolve(sys.getCurrentDirectory(), filePath)
         // external lib are transitive sources thus not listed in the inputs map reported by bazel.
-        if (!filesystemTree.fileExists(path.relative(execRoot, filePath)) && !isExternalLib(filePath) && !outputs.has(relative)) {
-            throw new Error(`tsc tried to read file (${filePath}) that wasn't an input to it.`);
-        }
-        return ts.sys.readFile(filePath, encoding);
-    }
-
-    function directoryExists(directoryPath) {
-        if (!directoryPath.startsWith(bin)) {
-            return false;
-        }
-        return filesystemTree.directoryExists(path.relative(execRoot, directoryPath));
-    }
-
-    function getDirectories(directoryPath) {
-        return filesystemTree.getDirectories(path.relative(execRoot, directoryPath));
-    }
-
-    function fileExists(filePath) {
-        return filesystemTree.fileExists(path.relative(execRoot, filePath));
-    }
-
-    
-    function readDirectory(directoryPath, extensions, exclude, include, depth) {
-        return filesystemTree.readDirectory(
-            path.relative(execRoot, directoryPath), 
-            extensions, exclude, include, depth
-        ).map(p => path.isAbsolute(p) ? p : path.join(execRoot, p))
-    }
-
-    function invalidate(filePath, kind) {
-        if (filePath.endsWith('.params')) return;
-        debuglog(`invalidate ${filePath} : ${ts.FileWatcherEventKind[kind]}`);
-
-        if (kind === ts.FileWatcherEventKind.Created) {
-            filesystemTree.add(filePath);
-        } else if (kind === ts.FileWatcherEventKind.Deleted) {
-            filesystemTree.remove(filePath);
-        } else {
-            filesystemTree.update(filePath);
-        }
+        // if (!filesystemTree.fileExists(relative) && !isExternalLib(filePath) && !outputs.has(filePath)) {
+        //     throw new Error(`tsc tried to read file (${filePath}) that wasn't an input to it.`);
+        // }
+        return ts.sys.readFile(path.join(execRoot, filePath), encoding);
     }
 
     function watchDirectory(directoryPath, callback, recursive, options) {
-        // since rules_js runs everything under bazel-out we shouldn't care about anything outside of it.
-        if (!directoryPath.startsWith(execRoot)) {
-            return { close: noop };
-        }
         const close = filesystemTree.watchDirectory(
-            path.relative(execRoot, directoryPath),
-            (input) => callback(path.join(execRoot, input)),
+            directoryPath,
+            (input) => watchEventQueue.push([callback, path.join("/", input)]),
```

Comment:
calling the callback immediately helped tsc to make wrong decisions when resolving modules. 

TODO: explain